packages/components/input-otp/src/use-input-otp.ts

```diff
@@ -223,6 +223,12 @@ export function useInputOtp(originalProps: UseInputOtpProps) {
     [baseDomRef, slots, baseStyles, isDisabled, isInvalid, isRequired, isReadOnly, value, length],
   );
 
+  const isNumeric = (pattern: string) => {
+    const numericPattern = /(^|\W)[0-9](\W|$)/;
+
+    return numericPattern.test(pattern) && !/[^\d\^$\[\]\(\)\*\+\-\.\|]/.test(pattern);
+  };
+
```

Comment:
@macci001, please move this to the shared-utils package, Also, ensure that the pattern string is validated to confirm it exists (i.e., it should not be null).

Reply:
Moved the check function to shared-utils.
Also, I think `allowedKeys` would not be null as we are setting the default value in L110